backend/utils.py

```diff
@@ -91,7 +92,21 @@ def wrapper() -> T:
 class AsyncProcessPoolExecutor(ProcessPoolExecutor):
     @staticmethod
     def _async_to_sync(__fn, *args, **kwargs):
-        return asyncio.run(__fn(*args, **kwargs))
+        try:
+            return asyncio.run(__fn(*args, **kwargs))
+        except Exception as e:
+            return Exception(f"Task failed: {str(e)}")
 
     def submit(self, __fn, *args, **kwargs):
-        return super().submit(self._async_to_sync, __fn, *args, **kwargs)
+        future = super().submit(self._async_to_sync, __fn, *args, **kwargs)
+        asyncio.create_task(self._check_future(future))
+        return future
+
+    async def _check_future(self, future):
+        try:
+            result = await asyncio.wrap_future(future)
+            if isinstance(result, Exception):
+                logger.error(f"Task in process pool failed: {result}")
+                # You could implement additional error handling here
+        except Exception as e:
+            logger.error(f"Error checking future in process pool: {e}")
```

Comment:
I would like to simplify a few things here
---
I was trying to get to the reason why the event loop closes in the first place, because I cannot reproduce it minimally

```
import time
import asyncio
from concurrent.futures import Executor, ProcessPoolExecutor

class AsyncProcessPoolExecutor(ProcessPoolExecutor):
    @staticmethod
    def _async_to_sync(__fn, *args, **kwargs):
        return asyncio.run(__fn(*args, **kwargs))

    def submit(self, __fn, *args, **kwargs):
        return super().submit(self._async_to_sync, __fn, *args, **kwargs)


pool = AsyncProcessPoolExecutor(max_workers=1)

async def crash():
    raise Exception("Crashed!")


async def main():
    futures = []
    for i in range(10):
        print("Submitting ... ")
        futures.append(pool.submit(crash))
        time.sleep(1)
    for fut in futures:
        print(f"{fut.exception()!r}")

if __name__ == "__main__":
    asyncio.run(main())
```

Let me quickly reproduce this with the whole setup